### PR TITLE
Fix additional kernels being triggered in CuteDSL MQA logits path

### PR DESCRIPTION
**Pull request by b8zhong**
It's a bug after refactoring in https://github.com/sgl-project/sglang/pull/25220

Extra kernels are dropped, and new results look inline with original PR results
Screenshot 2026-07-07 at 9 59 17 AM

It's very difficult to seperate out the indexer code out of DSA backend, since DSAMetadata is per forward (what we need), vs indexer is per layer

```
python test/registered/kernels/test_cute_dsl_fp8_paged_mqa_logits.py
============================================================================================== test session starts ==============================================================================================
platform linux -- Python 3.12.3, pytest-9.0.3, pluggy-1.6.0
rootdir: /sgl-workspace/sglang/test
configfile: pytest.ini
plugins: anyio-4.14.1, typeguard-4.5.2
collected 192 items                                                                                                                                                                                             

test/registered/kernels/test_cute_dsl_fp8_paged_mqa_logits.py ........................................................................................................................................... [ 72%]
.....................................................                                                                                                                                                     [100%]

=============================================================================================== warnings summary ================================================================================================
../../usr/local/lib/python3.12/dist-packages/_pytest/config/__init__.py:1309
../../usr/local/lib/python3.12/dist-packages/_pytest/config/__init__.py:1309
  /usr/local/lib/python3.12/dist-packages/_pytest/config/__init__.py:1309: PytestAssertRewriteWarning: Module already imported so cannot be rewritten; anyio
    self._mark_plugins_for_rewrite(hook, disable_autoload)

../../usr/local/lib/python3.12/dist-packages/_pytest/config/__init__.py:1434
  /usr/local/lib/python3.12/dist-packages/_pytest/config/__init__.py:1434: PytestConfigWarning: Unknown config option: asyncio_mode
  
    self._warn_or_fail_if_strict(f"Unknown config option: {key}\n")

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
================================================================================== 192 passed, 3 warnings in 78.03s (0:01:18) ===================================================================================
sys:1: DeprecationWarning: builtin type swigvarlink has no __module__ attribute
```

---
### CI States

Latest PR Test (Base): :x: [Run #29106549734](https://github.com/sgl-project/sglang/actions/runs/29106549734)
Latest PR Test (Extra): :x: [Run #29106549567](https://github.com/sgl-project/sglang/actions/runs/29106549567)
Latest PR Test (AMD ROCm 10): :heavy_minus_sign: **No AMD PR run found for this commit**.

### CODE REVIEW

**Comment by b8zhong** [comment]
/rerun-test test/registered/models_e2e/test_dsa_glm52_nvfp4_tp_mtp.py

**Comment by Fridge003** [comment]
/rerun-test test/registered/kernels/test_cute_dsl_fp8_paged_mqa_logits.py

**Comment by b8zhong** [comment]
Hi @trevor-m , we actually already integrated the FP8 version, are you looking for the FP4 CuteDSL one? I'm wondering if it's for use with DSV4. When testing GLM-5.2, the E2E gain is not very significant (~1%) since indexer is a small portion
